geopandas/tests/test_geodataframe.py

```diff
@@ -593,18 +593,39 @@ def test_from_feature_collection(self):
             "name": ["a", "b", "c"],
             "lat": [45, 46, 47.5],
             "lon": [-120, -121.2, -122.9],
+            "id": ["0", "1", "2"],
         }
 
         df = pd.DataFrame(data)
         geometry = [Point(xy) for xy in zip(df["lon"], df["lat"])]
         gdf = GeoDataFrame(df, geometry=geometry)
         # from_features returns sorted columns
-        expected = gdf[["geometry", "lat", "lon", "name"]]
+        expected = gdf[["geometry", "id", "lat", "lon", "name"]]
 
         # test FeatureCollection
         res = GeoDataFrame.from_features(gdf.__geo_interface__)
         assert_frame_equal(res, expected)
 
+        # test id as index
+        res = GeoDataFrame.from_features(gdf.__geo_interface__, id_as_index=True)
+        assert_frame_equal(res, expected.set_index("id"))
+
+        # test id field not present in FeatureCollection
+        feature_coll1 = gdf.__geo_interface__.copy()
+        for feat in feature_coll1["features"]:
+            del feat["id"]
+            del feat["properties"]["id"]
+        with pytest.warns(UserWarning):
```

Comment:
```suggestion
        with pytest.warns(UserWarning, match="Argument `id_as_index `is set to True"):
```

With `match` we can check the actual warning. The same applies to the other one below.